File: DeviseAlgorithm/test_2/Fibonacci/Divide/FibnacciDAC.c
/*
   斐波那契数分治法C语言实现
*/

#include<stdio.h>

/*分治法*/ 
int DAC_f(int n){ 

	if(n==1||n==2){
		return 1;}
	else{
		return DAC_f(n-1)+DAC_f(n-2);
	}
}
int main(){
	int n;
	printf("请输入斐波那契数列的项数: ");
	scanf("%d",&n);
	printf("DAC_f(%d)=%d",n,DAC_f(n));
	n=0;
	return 0;
}

File: DeviseAlgorithm/test_2/Fibonacci/Divide/FibnacciDAC.py
# 分治法解决斐波那契数列问题Python
# 学号：20201050470

def DAC_f(n):
    if n==1 or n==2:
        return 1
    else:
        return DAC_f(n-1)+DAC_f(n-2)

if __name__ == '__main__':
    n = (int)(input("斐波那契数列长度："))
    print("\n结果为:",DAC_f(n))